runtime/compiler/x/codegen/J9AheadOfTimeCompile.cpp

```diff
@@ -155,6 +155,31 @@ uint8_t *J9::X86::AheadOfTimeCompile::initializeAOTRelocationHeader(TR::Iterated
          }
         break;
 
+      case TR_BlockFrequency:
+         {
+         TR_RelocationRecordBlockFrequency *ptRecord = reinterpret_cast<TR_RelocationRecordBlockFrequency *>(reloRecord);
+         TR_PersistentProfileInfo *profileInfo = TR::comp()->getRecompilationInfo()->getProfileInfo();
+
+         TR_ASSERT(NULL != profileInfo, "PersistentProfileInfo not found when creating relocation record for block frequency\n");
+         if (NULL == profileInfo)
+            {
+            self()->comp()->failCompilation<J9::AOTRelocationRecordGenerationFailure>("AOT header initialization can't find profile info");
+            }
+
+         TR_BlockFrequencyInfo *blockFrequencyInfo = profileInfo->getBlockFrequencyInfo();
+         TR_ASSERT(NULL != blockFrequencyInfo, "BlockFrequencyInfo not found when creating relocation record for block frequency\n");
+         if (NULL == blockFrequencyInfo)
+            {
+            self()->comp()->failCompilation<J9::AOTRelocationRecordGenerationFailure>("AOT header initialization can't find block frequency info");
+            }
+
+         uintptr_t frequencyArrayBase = reinterpret_cast<uintptr_t>(blockFrequencyInfo->getFrequencyArrayBase());
+         uintptr_t frequencyPtr = reinterpret_cast<uintptr_t>(relocation->getTargetAddress());
+         ptRecord->setFrequencyOffset(reloTarget, frequencyPtr - frequencyArrayBase);
+         cursor = relocation->getRelocationData() + TR_RelocationRecord::getSizeOfAOTRelocationHeader(static_cast<TR_RelocationRecordType>(targetKind));
+         }
+         break;
```

Comment:
Why can't this be in the common `runtime/compiler/codegen/J9AheadOfTimeCompile.cpp` file?

Reply:
Only reason is I need to set relo flags in p and not in x, so they are slightly different. I think I can use the implementation for p  as the common code and when generating the relocation I can set relo flag to 0 in x.

Reply:
Yeah it makes sense that p had its own implementation, but the x implementation was the common one, so it could be moved to the common file. The way the infra works is that it will look at the platform specific implementation first, so even if you move the x implementation to the common file, on P it will still run the P implementation.

That said, if you change all platforms to use `TR_RelocationRecordInformation` then you won't need a separate version for P.

Reply:
> if you change all platforms to use TR_RelocationRecordInformation then you won't need a separate version for P

yup, that's the plan.